editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts

```diff
@@ -51,17 +51,13 @@ export function runGridRearrangeMove(
   selectedElement: ElementPath,
   jsxMetadata: ElementInstanceMetadataMap,
   interactionData: DragInteractionData,
+  grid: ElementInstanceMetadata,
 ): CanvasCommand[] {
   if (interactionData.drag == null) {
     return []
   }
 
-  const parentGridPath = EP.parentPath(selectedElement)
-  const grid = MetadataUtils.findElementByElementPath(jsxMetadata, parentGridPath)
-
-  if (grid == null) {
-    return []
-  }
+  const isReparent = !EP.pathsEqual(EP.parentPath(selectedElement), grid.elementPath)
```

Comment:
I think this is equivalent:
```suggestion
  const isReparent = !EP.isParentOf(grid.elementPath, selectedElement)
```